packages/upset/src/components/Rows/AggregateRow.tsx

```diff
@@ -66,14 +65,16 @@ export const AggregateRow: FC<Props> = ({ aggregateRow }) => {
   return (
     <g
       onMouseMove={(e) => e.stopPropagation()}
-      onClick={() => aggregateRow && (setCurrentIntersectionAtom(aggregateRow))}
+      onClick={() => aggregateRow && 
+        (currentIntersection?.id === aggregateRow.id ? 
+          actions.setSelected(null) : actions.setSelected(aggregateRow))}
       css={mousePointer}
     >
       <g transform={translate(aggregateRow.level === 2 ? secondLevelXOffset : 2, 0)}>
         <rect
           transform={translate(0, 2)}
           css={
-            (currentIntersection !== null && currentIntersection.id === aggregateRow.id) &&
+            (currentIntersection?.id === aggregateRow.id) &&
```

Comment:
does this throw an error if `currentIntersection === null` or does it simply fail the condition?

Reply:
That's the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) which returns `undefined` if the object we're trying to access is `null` or `undefined`. The previous `null` check was breaking because `currentIntersection` can now be `null` or `undefined`. (Side note; I'm not exactly sure why that is, but I figured tracking it down was more time consuming and less robust than ensuring all callers can handle `undefined`, and actually requiring it in the `currentIntersectionAtom` return type). So this simply fails the condition if `currentIntersection === null` (or `undefined`).